dali/operators/reader/loader/file_label_loader.h

```diff
@@ -47,19 +46,58 @@ class DLL_PUBLIC FileLabelLoader : public Loader<CPUBackend, ImageLabelWrapper>
  public:
   explicit inline FileLabelLoader(
     const OpSpec& spec,
-    vector<std::pair<string, int>> image_label_pairs = std::vector<std::pair<string, int>>(),
     bool shuffle_after_epoch = false)
     : Loader<CPUBackend, ImageLabelWrapper>(spec),
-      file_root_(spec.GetArgument<string>("file_root")),
-      file_list_(),
-      image_label_pairs_(std::move(image_label_pairs)),
       shuffle_after_epoch_(shuffle_after_epoch),
       current_index_(0),
       current_epoch_(0) {
 
-      // COCO doesn't have this argument, other inheriting class may not as well
-      if (spec.HasArgument("file_list")) {
+      vector<string> files;
+      vector<int> labels;
+
+      has_files_arg_ = spec.TryGetRepeatedArgument(files, "files");
+      has_labels_arg_ = spec.TryGetRepeatedArgument(labels, "labels");
+      has_file_list_arg_ = spec.TryGetArgument(file_list_, "file_list");
+      has_file_root_arg_ = spec.TryGetArgument(file_root_, "file_root");
+
+      DALI_ENFORCE(has_file_root_arg_ || has_files_arg_ || has_file_list_arg_,
+        "``file_root`` argument is required when not using ``files`` or ``file_list``.");
+
+      DALI_ENFORCE(has_files_arg_ + has_file_list_arg_ <= 1,
+        "File paths can be provided through ``files`` or ``file_list`` but not both.");
+
+      DALI_ENFORCE(has_files_arg_ || !has_labels_arg_,
+        "The argument ``labels`` is valid only when file paths "
+        "are provided as ``files`` argument.");
+
+      if (has_file_list_arg_) {
         file_list_ = spec.GetArgument<string>("file_list");
```

Comment:
Don't you already have the file_list_ from line 60?

Reply:
True.